apps/src/StudioApp.js

```diff
@@ -479,12 +479,25 @@ StudioApp.prototype.init = function (config) {
   this.alertIfAbusiveProject('#codeWorkspace');
 
   if (this.share && config.shareWarningInfo) {
-    shareWarnings.checkSharedAppWarnings({
-      channelId: config.channel,
-      isSignedIn: config.isSignedIn,
-      hasDataAPIs: config.shareWarningInfo.hasDataAPIs,
-      onWarningsComplete: config.shareWarningInfo.onWarningsComplete
-    });
+    var showWarnings = function () {
+      shareWarnings.checkSharedAppWarnings({
+        channelId: config.channel,
+        isSignedIn: config.isSignedIn,
+        hasDataAPIs: config.shareWarningInfo.hasDataAPIs,
+        onWarningsComplete: config.shareWarningInfo.onWarningsComplete
+      });
+    };
+    if (config.level.iframeEmbed) {
+      // temporarily replace startIFrameEmbeddedApp with a call to showWarnings
+      // this is kind of wonky IMO, would appreciate suggestions for how I ought
+      // to do this.
+      this.startIFrameEmbeddedApp = (originalFunc => function () {
+        showWarnings();
+        this.startIFrameEmbeddedApp = originalFunc;
+      })(this.startIFrameEmbeddedApp);
+    } else {
```

Comment:
well, the problem is that `startIFrameEmbeddedApp` gets called from `ApplabVisualizationColumn`, which doesn't have a reference to the config. So I can't pass `config` to `startIFrameEmbeddedApp`